scipy/optimize/lbfgsb.py

```diff
@@ -250,6 +251,10 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
         bounds = [(None, None)] * n
     if len(bounds) != n:
         raise ValueError('length of x0 != length of bounds')
+    # unbounded variables must use None, not +-inf, for optimizer to work properly
+    l = [None if l == -np.inf else l for l, _ in bounds]
+    u = [None if u == np.inf else u for _, u in bounds]
+    bounds = zip(l, u)
```

Comment:
Somewhat better would be to construct the `bounds` array in a single
list comprehension.